### PR TITLE
python3Packages.pytorch: 1.8.1 -> 1.9.0

### PR DESCRIPTION
**Pull request by danieldk**
###### Motivation for this change

Changelog:

https://github.com/pytorch/pytorch/releases/tag/v1.9.0

Tested build against CUDA :heavy_check_mark: .

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 127094` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages marked as broken and skipped:
</summary>
<ul>
<li>python38Packages.pywick</li>
<li>python38Packages.shap</li>
<li>python38Packages.skorch</li>
<li>python39Packages.pywick</li>
<li>python39Packages.skorch</li>
</ul>
</details>
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>tts: log was empty</li>
  </ul>
</details>
<details>

  <summary>3 packages failed to build and are new build failures:</summary>
  <ul>
<li>python38Packages.ignite: <a href="https://termbin.com/3x62">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/d9c6be95d61a211db3893adb68098014/raw/index.html">pretty log </a></li>
<li>python38Packages.pytorchWithCuda: log was empty</li>
<li>python39Packages.pytorchWithCuda: log was empty</li>
  </ul>
</details>
<details>
<summary>
22 packages built:
</summary>
<ul>
<li>python38Packages.pyro-ppl</li>
<li>python38Packages.pytorch (python38Packages.pytorchWithoutCuda)</li>
<li>python38Packages.pytorch-lightning</li>
<li>python38Packages.pytorch-metric-learning</li>
<li>python38Packages.rising</li>
<li>python38Packages.slicer</li>
<li>python38Packages.spacy-transformers</li>
<li>python38Packages.tensorboardx</li>
<li>python38Packages.test-tube</li>
<li>python38Packages.torchgpipe</li>
<li>python38Packages.torchvision</li>
<li>python39Packages.ignite</li>
<li>python39Packages.pyro-ppl</li>
<li>python39Packages.pytorch (python39Packages.pytorchWithoutCuda)</li>
<li>python39Packages.pytorch-lightning</li>
<li>python39Packages.pytorch-metric-learning</li>
<li>python39Packages.rising</li>
<li>python39Packages.slicer</li>
<li>python39Packages.tensorboardx</li>
<li>python39Packages.test-tube</li>
<li>python39Packages.torchgpipe</li>
<li>python39Packages.torchvision</li>
</ul>
</details>

File: pkgs/development/python-modules/pytorch/default.nix [hunk 1]
```diff
@@ -117,7 +117,7 @@ let
 in buildPythonPackage rec {
   pname = "pytorch";
   # Don't forget to update pytorch-bin to the same version.
```
Comment (jonringer) [inline pkgs/development/python-modules/pytorch/default.nix:119]: should we bump?

**Review by jonringer — APPROVED** [review]

File: pkgs/development/python-modules/pytorch/default.nix (hunk 1 above)
Comment (danieldk) [reply pkgs/development/python-modules/pytorch/default.nix:119]: Yeah, but I usually do separate PRs for them.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 127094` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages marked as broken and skipped:
</summary>
<ul>
<li>python38Packages.pywick</li>
<li>python38Packages.shap</li>
<li>python38Packages.skorch</li>
<li>python39Packages.pywick</li>
<li>python39Packages.skorch</li>
</ul>
</details>
<details>

  <summary>5 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>python39Packages.pytorch-metric-learning: log was empty</li>
<li>python39Packages.tensorboardx: log was empty</li>
<li>python39Packages.test-tube: log was empty</li>
<li>python39Packages.torchvision: log was empty</li>
<li>tts: log was empty</li>
  </ul>
</details>
<details>

  <summary>7 packages failed to build and are new build failures:</summary>
  <ul>
<li>python38Packages.ignite: log was empty</li>
<li>python38Packages.pytorch-metric-learning: log was empty</li>
<li>python38Packages.pytorchWithCuda: log was empty</li>
<li>python38Packages.rising: log was empty</li>
<li>python39Packages.ignite: log was empty</li>
<li>python39Packages.pytorchWithCuda: log was empty</li>
<li>python39Packages.torchgpipe: log was empty</li>
  </ul>
</details>
<details>
<summary>
14 packages built:
</summary>
<ul>
<li>python38Packages.pyro-ppl</li>
<li>python38Packages.pytorch (python38Packages.pytorchWithoutCuda)</li>
<li>python38Packages.pytorch-lightning</li>
<li>python38Packages.slicer</li>
<li>python38Packages.spacy-transformers</li>
<li>python38Packages.tensorboardx</li>
<li>python38Packages.test-tube</li>
<li>python38Packages.torchgpipe</li>
<li>python38Packages.torchvision</li>
<li>python39Packages.pyro-ppl</li>
<li>python39Packages.pytorch (python39Packages.pytorchWithoutCuda)</li>
<li>python39Packages.pytorch-lightning</li>
<li>python39Packages.rising</li>
<li>python39Packages.slicer</li>
</ul>
</details>

File: pkgs/development/python-modules/pytorch/default.nix (hunk 1 above)
Comment (danieldk) [reply pkgs/development/python-modules/pytorch/default.nix:119]: See #127305 and #127296.
